src/urllib3/connectionpool.py

```diff
@@ -1013,7 +1036,7 @@ def connection_from_url(url, **kw):
         return HTTPConnectionPool(host, port=port, **kw)
 
 
-def _normalize_host(host, scheme):
+def _normalize_host(host: Optional[str], scheme: Optional[str]) -> Optional[str]:
```

Comment:
`host` should be `str` here and return `str`?

Reply:
because `util.url._normalize_host` returns `Optional[str]`

Reply:
Ah gotcha, so a lot of this stems from there. Let's add an `@override` for `_normalize_host` where `None` returns `None` and `str` returns `str`? Then if we specify `str` here we don't have to return `Optional[str]`, does this solve the problem?